lib/dynamoid/transaction/executor.rb

```diff
@@ -0,0 +1,78 @@
+
+require 'dynamoid/transaction/action'
+require 'dynamoid/transaction/create'
+require 'dynamoid/transaction/delete'
+require 'dynamoid/transaction/update'
+require 'dynamoid/transaction/upsert'
+
+module Dynamoid
+  module Transaction
+    class Executor
+      def initialize(actions = [], options = {}, &block)
+        @skip_callbacks = !!options[:skip_callbacks]
+        @skip_validation = !!options[:skip_validation]
+        @actions = actions
+        instance_eval &block if block
+      end
+
+      def call
+        validate! unless @skip_validation
+        call_with_callbacks(@actions)
+      end
```

Comment:
It seems validation could be done twice? Here and in actions?

Reply:
Removed this.